### PR TITLE
Adding contributing.md first draft

### LINKED ISSUE #41: Add a Contributing.md file
Most projects that accept contributors include a contributing.md file which outlines the processes to follow for contributing to the project.  While we have the greater tableau.github.io that includes much of that information, calling out procedures locally would be useful for discoverability.  We should also include a link back to tableau.github.io's contributing page.

### PR DESCRIPTION
**Pull request by RussTheAerialist**
Fixes #41 
First draft of the contributing.md.  Please feel free to comment on wording, improvements, etc.

### CODE REVIEW

**Comment by benlower** [comment]
Let's get @lbrendanl and @jdomingu to weigh in too.

I like this. Would be good to add something like "If you are considering adding a new feature, we suggest starting a conversation about it first to identify any potential design or scheduling issues before jumping in."

**Comment by lbrendanl** [comment]
Would it be worth just briefly mentioning that issues and feature requests are done through GH issues and all other fixes are done through PRs?  I don't think it's uncommon for some to be unfamiliar with GitHub so it may be worth calling out.

**Comment by t8y8** [comment]
Looks good to me

**Comment by RussTheAerialist** [comment]
@benlower I just saw your second comment about starting a conversation.  I'll add something and send a new update real quick.

**Comment by benlower** [comment]
nice

File: contributing.md [hunk 1]
```diff
@@ -27,3 +27,7 @@ the limitations that you are running into, and provide us with a use case to kno
 
 For all other things, please submit a PR that includes the fix, documentation, or new code that you are trying to contribute.  More information on
 creating a PR can be found in the [github documentation](https://help.github.com/articles/creating-a-pull-request/)
+
+If the feature is complex or has multiple solutions that could be equally appropriate approaches, it would be helpful to file an issue to discuss the
+design trade-offs of each solution before implementing, to allow us to collective arrive at the best solution, which most likely exists in the middle
```
Comment (t8y8) [inline contributing.md]: `to collectively`

**Comment by RussTheAerialist** [comment]
yeah, I english good. Thanks @t8y8, fixed

**Comment by RussTheAerialist** [comment]
Anymore feedback on this one?  Or is it good to merge?

**Comment by jdomingu** [comment]
Ship it!

**Comment by benlower** [comment]
LGTM

**Pull request merged**